### PR TITLE
Various touchups to the pyenv setup

### PR DESCRIPTION
**Pull request by meatballhat**
including linking each installed version into `~travis/.pyenv/versions` for
compatibility with `pyenv` version-related commands, setting the global pyenv
version via `~travis/.pyenv/version`, and running `pyenv rehash` to ensure the
`~travis/.pyenv/shims` dir is populated.

/cc @audreyr @alex @dstufft

### CODE REVIEW

File: cookbooks/travis_python/recipes/pyenv.rb [hunk 1]
```diff
@@ -155,7 +167,7 @@
   end
 
   execute "install packages #{py}" do
-    command "#{virtualenv_root}/#{pyname}/bin/pip install --upgrade #{packages.join(' ')}"
+    command "#{virtualenv_root}/#{pyname}/bin/pip install -I -U #{packages.join(' ')}"
```
Comment (dstufft) [inline cookbooks/travis_python/recipes/pyenv.rb]: Why do you need the ``-I`` here? Nothing should already be installed in a brand new installation.

Comment (meatballhat) [inline cookbooks/travis_python/recipes/pyenv.rb]: I suppose I was concerned about system-level stuff muddying the waters.  I'll gladly back it out :+1:

**Comment by meatballhat** [comment]
Also /cc @coderanger :heart_eyes_cat:

**Comment by coderanger** [comment]
@meatballhat Hopefully in the near-ish future you'll be able to delete even more code, I need to get a `python-build` provider like I have for `poise-ruby-build`.

**Comment by meatballhat** [comment]
@coderanger :100: :sparkling_heart:

**Comment by meatballhat** [comment]
Eesh, I'll take a stab at this again later maybe.

**Pull request closed without merging**